test/router_SUITE.erl

```diff
@@ -782,6 +784,70 @@ join_test(Config) ->
     libp2p_swarm:stop(Swarm1),
     ok.
 
+us915_join_cf_list_test(Config) ->
+    AppKey = proplists:get_value(app_key, Config),
+    HotspotDir = proplists:get_value(base_dir, Config) ++ "/join_cf_list_test",
+    filelib:ensure_dir(HotspotDir),
+    RouterSwarm = blockchain_swarm:swarm(),
+    [Address | _] = libp2p_swarm:listen_addrs(RouterSwarm),
+    {Swarm, _} = test_utils:start_swarm(HotspotDir, no_this_is_patrick, 0),
+    PubKeyBin = libp2p_swarm:pubkey_bin(Swarm),
+
+    {ok, Stream} = libp2p_swarm:dial_framed_stream(
+        Swarm,
+        Address,
+        router_handler_test:version(),
+        router_handler_test,
+        [self(), PubKeyBin]
+    ),
+
+    %% Send join packet
+    DevNonce1 = crypto:strong_rand_bytes(2),
+    Stream ! {send, test_utils:join_packet(PubKeyBin, AppKey, DevNonce1, -100)},
+    timer:sleep(router_utils:join_timeout()),
+
+    %% Waiting for reply resp form router
+    {_NetID1, _DevAddr1, _DLSettings1, _RxDelay1, _NwkSKey1, _AppSKey1, CFList1} = test_utils:wait_for_join_resp(
+        PubKeyBin,
+        AppKey,
+        DevNonce1
+    ),
+    %% CFList should not be empty.
+    %% Has a channel mask setup
+    ?assertNotEqual(<<>>, CFList1),
+
+    %% Send another join to get a different cflist
+    DevNonce2 = crypto:strong_rand_bytes(2),
+    Stream ! {send, test_utils:join_packet(PubKeyBin, AppKey, DevNonce2, -100)},
+    timer:sleep(router_utils:join_timeout()),
+
+    {_NetID2, _DevAddr2, _DLSettings2, _RxDelay2, _NwkSKey2, _AppSKey2, CFList2} = test_utils:wait_for_join_resp(
+        PubKeyBin,
+        AppKey,
+        DevNonce2
+    ),
+    %% This CFList is empty. We don't know if the device is using lora version
+    %% older than 1.0.3 and might be choking on the cflist, so we'll send an
+    %% empty one and see if they can join with that. The device will still
+    %% receive a channel mask update in the first packets downlink after join.
```

Comment:
I would recommend you test this here to be sure?

Reply:
Added

Reply:
Well, I mean I was proposing we test we get the cflist in a downlink stiill unless this is already well tested elsewhere

Reply:
The link_adr_req command tested in some of the adr tests.
I beefed up this test to make sure the link_adr_req is turning on the same channels as the join cf_list.